dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/validation/ReferencesCheck.java

```diff
@@ -247,8 +283,50 @@ private void checkSharing( IdentifiableObject object, Preheat preheat,
     private boolean skipCheck( Class<?> klass )
     {
         return klass != null
-            && (EmbeddedObject.class.isAssignableFrom( klass )
-                || Period.class.isAssignableFrom( klass ) || PeriodType.class.isAssignableFrom( klass ));
+            && (Period.class.isAssignableFrom( klass ) || PeriodType.class.isAssignableFrom( klass ));
     }
 
+    private PreheatErrorReport createError( PreheatIdentifier identifier, ErrorCode errorCode, Object object,
+        IdentifiableObject refObject,
+        Property property, String... args )
+    {
+        if ( object instanceof IdentifiableObject identifiableObject )
+        {
+            return new PreheatErrorReport( identifier,
+                errorCode, identifiableObject, property, identifier.getIdentifiersWithName( refObject ),
+                identifier.getIdentifiersWithName( identifiableObject ), property.getName() );
+        }
+
+        return new PreheatErrorReport( identifier, object.getClass(), errorCode, property, args );
+    }
+
+    private boolean isShareable( ValidationContext context, Class<?> klass )
+    {
+        Schema schema = context.getSchemaService().getDynamicSchema( klass );
+        return schema != null && schema.isShareable();
+    }
```

Comment:
The `AclService` already has two `isShareable` methods. I think we can add this one, unless it is complicated to get access to the `AclService` here.

Reply:
right, i forgot about that. updated now.